src/main/java/InputHandler.java

```diff
@@ -0,0 +1,127 @@
+                    arr.add(newDeadline);
+                    printAddTaskMessage(newDeadline);
+                    return false;
+                } else {
+                    throw new DukeException(":( OOPS!!! The description of a deadline cannot be empty. Correct usage: deadline [task] /by [time]");
+                }
+            case "list":
+                if (splitInput.length == 1) {
+                    System.out.println("Here are the tasks in your list:");
+                    int i = 0;
+                    for (Task item : arr) {
+                        i += 1;
+                        if (item.isMark()) {
+                            System.out.println(i + ". " + item);
+                        } else {
+                            System.out.println(i + ". " + item);
+                        }
+                    }
+                    return false;
+                } else {
+                    throw new DukeException("Wrong usage of list! Correct usage: list");
+                }
+            case "mark":
+                if (splitInput.length == 2) {
+                    System.out.println("Nice! I've marked this task as done:\n");
+                    int idx = Integer.parseInt(splitInput[1]) - 1;
+                    Task taskToBeMarked = arr.get(idx);
+                    taskToBeMarked.markTask();
+                    return false;
+                } else {
+                    throw new DukeException("Wrong usage of mark! Correct usage: mark [index]");
+                }
+            case "unmark":
+                if (splitInput.length == 2) {
+                    System.out.println("OK, I've marked this task as not done yet:\n");
+                    int idx = Integer.parseInt(splitInput[1]) - 1;
+                    Task taskToBeUnmarked = arr.get(idx);
+                    taskToBeUnmarked.unmarkTask();
+                    return false;
+                } else {
+                    throw new DukeException("Wrong usage of unmark! Correct usage: unmark [index]");
+                }
+            case "delete":
+                if (splitInput.length == 2) {
+                    int idx = Integer.parseInt(splitInput[1]) - 1;
+                    Task taskToBeDeleted = arr.get(idx);
+                    arr.remove(idx);
+                    System.out.println("Noted. I've removed this task:\n" + taskToBeDeleted + "\nNow you have " + arr.size() + " tasks in the list");
+                    return false;
+                } else {
+                    throw new DukeException("Wrong usage of delete! Correct usage: delete [index]");
+                }
+            case "bye":
+                return true;
+            default:
+                throw new DukeException(":( OOPS!!! I'm sorry, but I don't know what that means! Possible commands: todo [task], event [task] /at [time],"
+                + " deadline [task] /by [time], mark [index], unmark [index], delete [index], bye");
+        }
+
+    }
```

Comment:
This method severely exceeds the recommended number of lines in a method (30). 
Recommendation:
Can try to abstract out the details of each case statement into a different method.